packages/kbn-unified-data-table/src/components/data_table_columns.tsx

```diff
@@ -125,9 +125,11 @@ function buildEuiGridColumn({
       : [];
   }
 
+  const columnType = dataViewField?.type ?? columnTypes?.[columnName];
```

Comment:
`columnTypes` were introduced only recently for ES|QL view to show correct icons next to fields. I don't think it's a regression.

It would be better to swap and base it on columns which are coming back from ES|QL request.

```suggestion
  const columnType = columnTypes?.[columnName] ?? dataViewField?.type;
```

Reply:
I remember we had a simpler way of computing these columns and I had tested this scenario when Peter added the in memory sorting but I might be wrong.